tools/gfx/vulkan/vk-device.cpp

```diff
@@ -1638,6 +1650,96 @@ Result DeviceImpl::createTextureResource(
             VK_IMAGE_LAYOUT_UNDEFINED,
             VK_IMAGE_LAYOUT_TRANSFER_DST_OPTIMAL);
 
+        if(desc.sampleDesc.numSamples != 1)
+        {
+            // Handle senario where texture is sampled. We cannot use
+            // a simple buffer copy for sampled textures. ClearColorImage
+            // is not data accurate but it is fine for testing & works.
+            FormatInfo formatInfo;
+            gfxGetFormatInfo(desc.format, &formatInfo);
+            uint32_t data = 0;
+            VkClearColorValue clearColor;
+            switch(formatInfo.channelType)
+            {
+            case SLANG_SCALAR_TYPE_INT32:
+                memcpy(&data, initData->data, sizeof(data));
+                break;
+            case SLANG_SCALAR_TYPE_UINT32:
+                memcpy(&data, initData->data, sizeof(data));
+            case SLANG_SCALAR_TYPE_INT64:
+            {
+                int32_t tmpData = int32_t(((int64_t*)initData->data)[0]);
+                memcpy(&data, &tmpData, sizeof(data));
+                break;
+            }
+            case SLANG_SCALAR_TYPE_UINT64:
+            {
+                uint32_t tmpData = uint32_t(((uint64_t*)initData->data)[0]);
+                memcpy(&data, &tmpData, sizeof(data));
+                break;
+            }
+            case SLANG_SCALAR_TYPE_FLOAT16:
+            {
+                float tmpData = HalfToFloat(((uint16_t*)initData->data)[0]);
+                memcpy(&data, &tmpData, sizeof(data));
+                break;
+            }
+            case SLANG_SCALAR_TYPE_FLOAT32:
+            {
+                float tmpData = float(((float*)initData->data)[0]);
+                memcpy(&data, &tmpData, sizeof(data));
```

Comment:
I think it will be easier if you just assign the values without going through "data".
```
for (int i =0; i < 4; ++i)
clearColor.float32[i] = *reinterpret_cast<float*>(initData->data);
```